packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/AmplifyPushNotificationsPlugin.kt

```diff
@@ -3,110 +3,250 @@
 
 package com.amazonaws.amplify.amplify_push_notifications
 
+import android.app.Activity
+import android.content.Context
 import android.content.Intent
+import android.content.SharedPreferences
 import androidx.annotation.NonNull
+import androidx.core.app.ActivityCompat
 import com.amazonaws.amplify.AtomicResult
-import com.google.android.gms.tasks.OnCompleteListener
+import com.amplifyframework.pushnotifications.pinpoint.utils.permissions.PermissionRequestResult
+import com.amplifyframework.pushnotifications.pinpoint.utils.permissions.PushNotificationPermission
 import com.google.firebase.messaging.FirebaseMessaging
-import com.google.firebase.messaging.RemoteMessage
 import io.flutter.Log
 import io.flutter.embedding.engine.plugins.FlutterPlugin
 import io.flutter.embedding.engine.plugins.activity.ActivityAware
 import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+import io.flutter.plugin.common.BinaryMessenger
 import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 import io.flutter.plugin.common.PluginRegistry
+import kotlinx.coroutines.*
+
+
+private const val TAG = "AmplifyPushNotificationsPlugin"
 
 /** AmplifyPushNotificationsPlugin */
-class AmplifyPushNotificationsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
+open class AmplifyPushNotificationsPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
     PluginRegistry.NewIntentListener {
+    private companion object {
+        const val TAG = "AmplifyPushNotificationsPlugin"
```

Comment:
I don't think you need this since you already have the TAG declared on this file and you don't access it statically anywhere else (I think?)

Reply:
right, I missed to remove the one defined for the file. It's in the object because I was following the convention from [AuthCognito](https://github.com/aws-amplify/amplify-flutter/blob/next/packages/auth/amplify_auth_cognito_android/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt#L140) but you are right on not wanting to use it statically.